### PR TITLE
Discover devices over the network

### PR DESCRIPTION
**Pull request by MatteoGheza**
I selected the wrong base when creating the PR.

TODOs:
- [x] Fix decorators
- [x] Add a decorator for adding devices after network discovery
- [x] Add discovered devices in the UI
- [x] Open source modal clicking "Add" in discovered devices list
- [x] Support for multiple addresses
- [x] Hide devices that are already configured

### CODE REVIEW

**Comment by MatteoGheza** [comment]
Closes #174 #228

File: UI/src/app/_components/settings/settings.component.ts [hunk 1]
```diff
@@ -161,6 +162,8 @@ export class SettingsComponent {
 	}
 
 	public navChanged(event: any) {
+		console.log(this.socketService.sourceTypes);
```
Comment (hrueger) [inline UI/src/app/_components/settings/settings.component.ts:165]: debug line?

File: UI/src/app/_components/settings/settings.component.ts [hunk 2]
```diff
@@ -161,6 +162,8 @@ export class SettingsComponent {
...
+		(window as any).socketService = this.socketService;
```
Comment (hrueger) [inline UI/src/app/_components/settings/settings.component.ts:166]: why do we need to share this service using the window object?
I can't find where it is used...

**Review by hrueger — APPROVED** [review]
lgtm, although I haven't tested it yet. I will do that, it looks really useful.
Nice work with the decorators 👍

**Comment by MatteoGheza** [comment]
I'm sorry @hrueger I forgot again to remove debug lines... 😄

**Comment by hrueger** [comment]
Never mind ;-)

**Pull request merged**